### ISSUE TITLE
ItemStack#builder#fromContainer not correctly deserializing HOCON containers

### ISSUE THREAD

**Issue by Zerthick** [SpongePowered/Sponge#2094]
**I am currently running**
- SpongeVanilla version: 1.12.2-7.1.0-BETA-100

**Issue Description**
As the title says, when attempting to recreate an itemstack with custom data manipulators attached from a HOCON string, those custom manipulators are lost. 

Serialization/Deserialization Code:
```java
public static String serializeSnapShot(ItemStackSnapshot snapshot) throws IOException {

    DataContainer container = snapshot.toContainer();
    StringWriter stringWriter = new StringWriter();

    DataFormats.HOCON.writeTo(stringWriter, container);

    return stringWriter.toString();
}

public static ItemStackSnapshot deserializeSnapShot(String serializedSnapshot) throws IOException {

    DataContainer container = DataFormats.HOCON.read(serializedSnapshot);

    return ItemStack.builder().fromContainer(container).build().createSnapshot();
}
```

The string I'm attempting to deserialize (which was created by the above code) is as follows:
```
ContentVersion=2
Count=1
Data=[
    {
        ContentVersion=2
        ManipulatorData {
            CommandBookCommands=[
                "$time set day"
            ]
            CommandBookUses=-1
            ContentVersion=1
        }
        ManipulatorId="commandbooks:command_book"
    }
]
ItemType="minecraft:enchanted_book"
UnsafeDamage=0
UnsafeData {
    display {
        Lore=[
            "Uses: ∞"
        ]
        Name="§b§o[Day]"
    }
}
```

**Comment by Shiunin** [comment]
If it helps at all I am also experiencing this issue and can provide steps to reproduce if needed. I'm on this version:
![image](https://user-images.githubusercontent.com/44011334/55523472-4d847c00-563e-11e9-94a1-2b0cb5d8af7b.png)

**Comment by randombyte-developer** [comment]
Still reproducible on SV `1.12.2-7.1.5`.

Some debugging from my side got me to this:

[This](https://github.com/SpongePowered/SpongeCommon/blob/c6724ece4e32d228c828a8190b6fae42b1adede9/src/main/java/org/spongepowered/common/data/builder/item/SpongeItemStackSnapshotBuilder.java#L69) shouldn't call `DataView#getViewList(...)`. The data view was built/translated from a config node by Sponge. The Data node is in the container. But the `getViewList()` doesn't return anything, because the Data node in there isn't a DataView.

That's the getViewList method for reference: https://github.com/SpongePowered/SpongeCommon/blob/c6724ece4e32d228c828a8190b6fae42b1adede9/src/main/java/org/spongepowered/common/data/MemoryDataView.java#L709

Calling something more generic works in this case, like `MemoryDataView#get()`.

But in the end I don't know if the translation from the config node to the DataView went wrong, or if the "wrong" 'get'-method is called.

**Comment by gabizou** [comment]
Should be fixed with the merge of #2275.
